### PR TITLE
fix(traffic_control): Install traffic control on AMAZON2 image

### PR DESCRIPTION
**Pull request by aleksbykov**
AMAZON2 image doesn't have traffic control package installed.
Check and install package if needed

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -693,6 +693,14 @@ def traffic_control(self, tcconfig_params=None):
             tc_command = LOCAL_CMD_RUNNER.run("tcset eth1 {} --tc-command".format(tcconfig_params)).stdout
             self.remoter.run('sudo bash -cxe "%s"' % tc_command)
 
+    def install_traffic_control(self):
```
Comment (bentsi) [inline sdcm/cluster_aws.py:696]: If this is related to Amazon Linux only, better to install only for it.
```
node.distro.is_amazon2
```

File: sdcm/cluster_aws.py [hunk 2]
```diff
@@ -693,6 +693,14 @@ def traffic_control(self, tcconfig_params=None):
...
+        self.log.debug("Check that traffic_control is installed")
+        res = self.remoter.run("/sbin/tc -h", ignore_status=True)
+        if not res.ok:
+            self.log.debug("traffic control package is not installed. Installing...")
+            self.pkg_install("iproute-tc")
```
Comment (enaydanov) [inline sdcm/cluster_aws.py:701]: Wow, first usage of this method since 2y after it's introduction :-)

There is no advantage over `self.remoter.run("yum install -y iproute-tc")` in the way you call it.

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -683,16 +683,22 @@ def traffic_control(self, tcconfig_params=None):
...
-
         if tcconfig_params is None:
             tc_command = LOCAL_CMD_RUNNER.run("tcdel eth1 --tc-command", ignore_status=True).stdout
             self.remoter.run('sudo bash -cxe "%s"' % tc_command, ignore_status=True)
         else:
             tc_command = LOCAL_CMD_RUNNER.run("tcset eth1 {} --tc-command".format(tcconfig_params)).stdout
             self.remoter.run('sudo bash -cxe "%s"' % tc_command)
 
+    def install_traffic_control(self):
+        self.remoter.run("sudo modprobe sch_netem")
```
Comment (bentsi) [inline sdcm/cluster_aws.py:694]: why did you move the modprobe here? the node can be restarted in between network nemesis and the module won't be loaded.

Comment (aleksbykov) [reply sdcm/cluster_aws.py:694]: it runs 3 times for 1 nemesis each time when the traffic_control is called.
I call install_traffic_control each time in nemesis at the beginning

Comment (amoskong) [reply sdcm/cluster_aws.py:694]: It sounds safe to load the module by modprobe everytime.

But we only need to prepare once for the whole job with following steps:

- load the module, make it work before restart
    - sudo modprobe sch_netem
- install tc
- set to load module after restart
    - echo sch_netem |sudo tee /etc/modules-load.d/tc.conf

**Comment by amoskong** [comment]
We can integrate the prepare inside traffic_control(), or putting it as a decorator.

```
diff --git a/sdcm/cluster_aws.py b/sdcm/cluster_aws.py
index a87bf0e5..1dc360e4 100644
--- a/sdcm/cluster_aws.py
+++ b/sdcm/cluster_aws.py
@@ -684,7 +684,18 @@ class AWSNode(cluster.BaseNode):
         :return: None
         """
 
-        self.remoter.run("sudo modprobe sch_netem")
+        if not self.tc_is_prepared:
+            self.remoter.run("sudo modprobe sch_netem")
+            self.remoter.run("echo sch_netem |sudo tee /etc/modules-load.d/tc.conf")
+
+            if self.distro.is_amazon2:
+                self.log.debug("Installing iproute-tc package for AMAZON2")
+                self.remoter.run("sudo yum install -y iproute-tc", ignore_status=True)
+
+            if not self.remoter.run("/sbin/tc -h", ignore_status=True).ok:
+                raise UnsupportedNemesis("Traffic control package not installed on system")
+
+        self.tc_is_prepared = True
 
         if tcconfig_params is None:
             tc_command = LOCAL_CMD_RUNNER.run("tcdel eth1 --tc-command", ignore_status=True).stdout
```

File: sdcm/cluster_aws.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/cluster_aws.py:694]: @amoskong why it could be safer? the nemesises which use it are disruptive, so it will not be running in parallel with any reboot/hard reboot nemesis. So i think it will be enough to load it on nemesis start.

Comment (amoskong) [reply sdcm/cluster_aws.py:694]: > @amoskong why it could be safer? the nemesises which use it are disruptive, so it will not be running in parallel with any reboot/hard reboot nemesis. So i think it will be enough to load it on nemesis start.

Your current solution is safe.

Comment (bentsi) [reply sdcm/cluster_aws.py:694]: the simplest solution is to keep the module loading in the place where it was before traffic_control(). Why don't you want to keep it there?

Comment (aleksbykov) [reply sdcm/cluster_aws.py]: The idea was to have all commands related to installation in one function.
Traffic_control is called 3 times during one nemesis, So i didn't see requirement to execute command to load module each time.

Return as you recommended
